core/src/main/scala/sttp/tapir/PathComponent.scala

```diff
@@ -0,0 +1,15 @@
+package sttp.tapir
+
+import sttp.model.Method
+
+sealed trait PathComponent
+
+case object PathVariableSegment extends PathComponent
```

Comment:
I'd move these inside `ShadowedEndpointChecker` and make them private as they are used only internally in this class (nothing end-user-facing)